fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/ReferencesRdfContext.java

```diff
@@ -48,21 +57,36 @@ public ReferencesRdfContext(final FedoraResource resource,
         throws RepositoryException {
         super(resource, idTranslator);
         property2triple = new PropertyToTriple(resource.getNode().getSession(), idTranslator);
-        concat(putStrongReferencePropertiesIntoContext());
-        concat(putWeakReferencePropertiesIntoContext());
+        session = resource.getNode().getSession();
+        putReferencesIntoContext(resource.getNode().getWeakReferences());
+        putReferencesIntoContext(resource.getNode().getReferences());
     }
 
-    private Iterator<Triple> putWeakReferencePropertiesIntoContext() throws RepositoryException {
-        final Iterator<Property> properties = resource().getNode().getWeakReferences();
+    private void putReferencesIntoContext(final Iterator<Property> properties) throws RepositoryException {
+        while (properties.hasNext()) {
```

Comment:
Yes, basically it's iterating over the properties of the nodes that refer to it.  For each of them, it might return nothing if the reference isn't from a proxy, or it might return several triples if that proxy is in several containers.

Now that I put it that way, it sounds like I need to break down the logic into a chain of filtering and transforming steps.